crates/tracing_macros/src/autotrace.rs

```diff
@@ -21,16 +47,53 @@ impl Autotrace {
 }
 
 impl Autotrace {
-    fn rewrite_block(name: String, block: syn::Block) -> syn::Block {
-        syn::parse(TokenStream::from(quote! {
-            {
-                let __autotrace_guard = ::holochain_tracing::push_span_with(
-                    |span| span.child(#name)
-                );
-                #block
-            }
-        }))
-        .expect("Couldn't parse statement when rewriting block")
+    fn rewrite_block(
+        app_name: &String,
+        license_key: &Option<String>,
+        name: String,
+        block: syn::Block,
+    ) -> syn::Block {
+        let new_block = license_key
+            .as_ref()
+            .map(|license_key| {
+                //structure of func created will replce old function but have new relic recording capabilities
+                quote! {
+                {
+                    let __autotrace_guard = ::holochain_tracing::push_span_with(
+                        |span| span.child(#name)
+                    );
+
+                    //if new relic is somehow down or the daemon is not running, the program should continue normally
+                    //stated away from combinators because of closure ownership
+                    if let Ok(live_app) = newrelic::App::new(#app_name, #license_key)
+                    {
+                        if let Ok(transaction) = live_app.non_web_transaction(#name)
+                        {
+                            #block
+                        }
+                        else
+                        {
+                            #block
+                        }
+                    }
+                    else
+                    {
+                        #block
+                    }
+                }}
+            })
+            .unwrap_or_else(|| {
```

Comment:
So it falls back to the previous autotracing implementation- interesting. Is there any reason to do both?

Reply:
If something happens ever happens when contacting the newrelic external services or if it is not configured to use new_relic I would like it to behave as normal